service/src/test/java/uk/gov/hmcts/reform/fpl/service/ManageDocumentServiceTest.java

```diff
@@ -815,6 +821,267 @@ void shouldNotRemoveHearingFurtherEvidenceBundleElementWhenDocumentsForThatHeari
+                        .respondentId(UUID.randomUUID())
+                        .supportingEvidenceBundle(supportingEvidenceBundleTwo)
+                        .build())))
+                .build();
+
+            List<Element<SupportingEvidenceBundle>> actualBundle
+                = manageDocumentService.getRespondentStatementFurtherEvidenceCollection(caseData, selectedRespondentId);
+
+            assertThat(actualBundle.get(0).getValue()).isEqualTo(SupportingEvidenceBundle.builder().build());
+        }
+
+        @Test
+        void shouldReturnEmptySupportingEvidenceDocumentsWhenCaseDoesNotContainRespondentStatements() {
+            CaseData caseData = CaseData.builder().build();
+
+            List<Element<SupportingEvidenceBundle>> actualBundle
+                = manageDocumentService.getRespondentStatementFurtherEvidenceCollection(caseData, selectedRespondentId);
+
+            assertThat(actualBundle.get(0).getValue()).isEqualTo(SupportingEvidenceBundle.builder().build());
+        }
+    }
+
+    @Nested
+    class GetUpdatedRespondentStatements {
+        UUID respondentOneId = UUID.randomUUID();
+        UUID respondentTwoId = UUID.randomUUID();
+        UUID respondentStatementId = UUID.randomUUID();
+        UUID supportingEvidenceBundleId = UUID.randomUUID();
+
+        @Test
+        void shouldUpdateExistingRespondentStatementsWithNewBundle() {
+            DynamicList respondentStatementDynamicList = buildRespondentStatementDynamicList();
+            List<Element<SupportingEvidenceBundle>> updatedBundle = buildSupportingEvidenceBundle();
+
+            CaseData caseData = CaseData.builder()
+                .respondents1(List.of(
+                    element(respondentOneId, Respondent.builder()
+                        .party(RespondentParty.builder()
+                            .firstName("David")
+                            .lastName("Stevenson")
+                            .build())
+                        .build()),
+                    element(Respondent.builder().build())))
+                .supportingEvidenceDocumentsTemp(updatedBundle)
+                .respondentStatementDynamicList(respondentStatementDynamicList)
+                .respondentStatements(newArrayList(
+                    element(respondentStatementId, RespondentStatement.builder()
+                        .respondentId(respondentOneId)
+                        .supportingEvidenceBundle(newArrayList(
+                            element(supportingEvidenceBundleId, SupportingEvidenceBundle.builder().build())
+                        ))
+                        .respondentName("David Stevenson")
+                        .build())))
+                .build();
+
+            List<Element<RespondentStatement>> updatedRespondentStatements =
+                manageDocumentService.getUpdatedRespondentStatements(caseData);
+
+            assertThat(updatedRespondentStatements.size()).isEqualTo(1);
```

Comment:
size assertion is not needed as next assertion covers it, best would be to use:
```
 assertThat(updatedRespondentStatements).containsExactly(
                element(...
 ```
 
 it gives better error on failure